tools/releaser/main.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2023 Uber Technologies, Inc.
+// Licensed under the Apache License, Version 2.0
+
+// releaser is a tool for managing part of the process to release a new version of bazel-zig-cc.
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/google/zopfli/go/zopfli"
+)
+
+var (
+	// Paths to be included to the release
+	_paths = []string{
+		"LICENSE",
+		"NOTICE",
+		"README.md",
+		"toolchain/*",
+	}
+
+	// regexp for valid tags
+	tagRegexp = regexp.MustCompile("^v([0-9]+)\\.([0-9]+)(\\.([0-9]+))(-rc([0-9]+))?$")
+
+	tagErr = errors.New("tag accepts the following formats: v1.0.0 v1.0.1-rc1")
+)
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer cancel()
+	if err := run(ctx, os.Stderr); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func log(msg string, format ...any) {
+	fmt.Fprintf(flag.CommandLine.Output(), msg+"\n", format...)
+}
+
+func run(ctx context.Context, stderr *os.File) error {
```

Comment:
```suggestion
func run(ctx context.Context, stderr io.Writer) error {
```
`io.Writer` is more general, so we can potentially pass in a buffer during a test.